pkg/snapstore/abs_snapstore.go

```diff
@@ -73,6 +76,21 @@ func NewABSSnapStore(config *brtypes.SnapstoreConfig) (*ABSSnapStore, error) {
 			TryTimeout: downloadTimeout,
 		}})
 	u, err := url.Parse(fmt.Sprintf("https://%s.%s", storageAccount, brtypes.AzureBlobStorageHostName))
+	// Check if backup-restore is running with the Azure Blob Storage Emulator - Azurite
+	if enableAzurite, ok := os.LookupEnv(envEnableAzurite); ok {
+		isEmulator, err := strconv.ParseBool(enableAzurite)
+		if err != nil {
+			return nil, fmt.Errorf("value %s for the environment variable %s could not be parsed", enableAzurite, envEnableAzurite)
+		}
+		if isEmulator {
+			if endpoint, ok := os.LookupEnv(azuriteEndpoint); ok {
+				// application protocol is set by the user of Azurite, not by backup-restore
+				u, err = url.Parse(fmt.Sprintf("%s/%s", endpoint, credentials.AccountName()))
+			} else {
+				return nil, fmt.Errorf("environment variable %s not set while the environment variable %s is set to true", azuriteEndpoint, envEnableAzurite)
+			}
+		}
+	}
```

Comment:
Could we move the logic for constructing the URL into a separate function? This would enhance the readability of `NewABSSnapStore`.


```
// NewABSSnapStore creates a new ABSSnapStore using a shared configuration and a specified bucket.
func NewABSSnapStore(config *brtypes.SnapstoreConfig) (*ABSSnapStore, error) {
	storageAccount, storageKey, err := getCredentials(getEnvPrefixString(config.IsSource))
	if err != nil {
		return nil, fmt.Errorf("failed to get credentials: %v", err)
	}

	credentials, err := azblob.NewSharedKeyCredential(storageAccount, storageKey)
	if err != nil {
		return nil, fmt.Errorf("failed to create shared key credentials: %v", err)
	}

	pipeline := azblob.NewPipeline(credentials, azblob.PipelineOptions{
		Retry: azblob.RetryOptions{
			TryTimeout: downloadTimeout,
		}})

	blobURL, err := constructBlobServiceURL(storageAccount, credentials)
	if err != nil {
		return nil, err
	}

	serviceURL := azblob.NewServiceURL(*blobURL, pipeline)
	containerURL := serviceURL.NewContainerURL(config.Container)

	return GetABSSnapstoreFromClient(config.Container, config.Prefix, config.TempDir, config.MaxParallelChunkUploads, config.MinChunkSize, &containerURL)
}

func constructBlobServiceURL(storageAccount string, credentials *azblob.SharedKeyCredential) (*url.URL, error) {
	defaultURL, err := url.Parse(fmt.Sprintf("https://%s.%s", storageAccount, brtypes.AzureBlobStorageHostName))
	if err != nil {
		return nil, fmt.Errorf("failed to parse default service URL: %v", err)
	}

	if emulatorEnabled, ok := os.LookupEnv(envEmulatorEnabled); ok {
		isEmulator, err := strconv.ParseBool(emulatorEnabled)
		if err != nil {
			return nil, fmt.Errorf("invalid value for %s: %s, error: %v", envEmulatorEnabled, emulatorEnabled, err)
		}

		if isEmulator {
			if endpoint, ok := os.LookupEnv(azuriteEndpoint); ok {
				return url.Parse(fmt.Sprintf("%s/%s", endpoint, credentials.AccountName()))
			}
			return nil, fmt.Errorf("%s environment variable not set while %s is true", azuriteEndpoint,envEmulatorEnabled )
		}
	}

	return defaultURL, nil
}
```

Reply:
Thanks for the suggestion @seshachalam-yv! The readability does improve. Will make the necessary changes.

Reply:
Addressed in [97fac6f](https://github.com/gardener/etcd-backup-restore/pull/699/commits/97fac6fb320902e7f44971717e2d75dd449e9570).